### PR TITLE
improve warning message about global namespace class

### LINKED ISSUE #417: DocFX currently only supports generating metadata with namespace defined
When I run ```docfx docfx.json --serve``` from the command line I get warnings for a couple of projects in my solution.

`Warning: No namespace is found in assembly XXX. DocFX currently only supports generating metadata with namespace defined.`

How do I fix this?

### PR DESCRIPTION
**Pull request by superyyrrzz**
fix #417 
@chenkennt @vwxyzh @ansyral @hellosnow @qinezh

### CODE REVIEW

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/Visitors/SymbolVisitorAdapter.cs [hunk 1]
```diff
@@ -130,9 +130,10 @@ public override MetadataItem VisitAssembly(IAssemblySymbol symbol)
             item.Type = MemberType.Assembly;
             _references = new Dictionary<string, ReferenceItem>();
             var namespaces = symbol.GlobalNamespace.GetNamespaceMembers().ToList();
-            if (namespaces.Count == 0)
+            var allMembers = symbol.GlobalNamespace.GetMembers().ToList();
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/Visitors/SymbolVisitorAdapter.cs]: how about getting the non namespace members directly?

**Comment by ansyral** [comment]
👍

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/Visitors/SymbolVisitorAdapter.cs [hunk 2]
```diff
@@ -130,9 +130,10 @@ public override MetadataItem VisitAssembly(IAssemblySymbol symbol)
...
+            if (namespaces.Count != allMembers.Count)
             {
-                Logger.LogWarning($"No namespace is found in assembly {symbol.MetadataName}. DocFX currently only supports generating metadata with namespace defined.");
+                Logger.LogWarning($"DocFX currently only supports generating metadata with namespace defined. The following types in assembly \"{symbol.MetadataName}\" will have no metadata generated: {string.Join(", ", allMembers.Except(namespaces).Select(m => m.MetadataName))}. ");
```
Comment (qinezh) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/Visitors/SymbolVisitorAdapter.cs]: how about using `\n{string.Join("\t")}` to make warning message more readable when there're too many non namespace members?

**Comment by hellosnow** [comment]
:shipit:

**Comment by vwxyzh** [comment]
👍

**Comment by qinezh** [comment]
:+1:

**Pull request merged**